internal/provider/virtual_cluster.go

```diff
@@ -5,8 +5,34 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+type apiKeyModel struct {
+	Name         types.String       `tfsdk:"name"`
+	Key          types.String       `tfsdk:"key"`
+	CreatedAt    types.String       `tfsdk:"created_at"`
+	AccessGrants []accessGrantModel `tfsdk:"access_grants"`
+}
+
+type accessGrantModel struct {
+	PrincipalKind types.String `tfsdk:"principal_kind"`
+	ResourceKind  types.String `tfsdk:"resource_kind"`
+	ResourceID    types.String `tfsdk:"resource_id"`
+}
+
 // virtualClusterModel maps virtual cluster schema data.
-type virtualClusterModel struct {
```

Comment:
Split this into two types so that I could change the data source schema without touching the resource schema in the same pr.